Docs updates -- Do not merge (#149)
Azure/azure-saas: docs/azure-saas-docs/content/en/quick-start.md

---
type: docs
title: "Quick Start"
linkTitle: "Quick Start"
weight: 10
description: "Getting Started with the Azure SaaS Dev Kit"
---

On this page, you will find instructions to run the dev kit in your local environment, how to deploy the solution to Azure, and where to put your application code to customize the solution.

> Tip: If you're new here and want to learn what this is, check out the [welcome page](..)

## 1. Setup Identity Framework

This project uses [Azure Active Directory B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/overview) for an IDP (Identity Provider). The first step in setting up this project is to configure a new B2C instance to house your local user accounts. You will also need to deploy the [Permissions API](../components/identity/permissions-service), as B2C will have a dependancy on it.

> To setup the Identity Framework, we have provided a powershell script [here]() that automates the setup for you. This powershell script will output a parameters file that you'll need to provide when deploying the solution to azure in step 2.b.

Author: Correct. No link yet. It should be finished soon.

After finishing the IDP setup, you may choose to either run the project locally first or immediately deploy the solution to Azure.
## 2.a. Running the Dev Kit in your local dev environment

- Install the latest version of [Visual Studio 2022](https://visualstudio.microsoft.com/vs/). You may also use Visual Studio Code, but the solution and projects are targetted at VS2022.
- Clone the repository `https://github.com/Azure/azure-saas.git` on to your dev machine.
- Open the `.sln` in the root of the repository. This solution includes all of the modules.
- Depending on the project you wish to run, you'll need to set some secrets to properly setup authentication with B2C. See the [App Settings](#app-settings) section below

### App Settings

> When running the project locally, you will need to set some app settings & secrets manually using the [.NET secrets manager](https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-6.0#secret-storage-in-the-development-environment). When deployed to azure (below), these secrets are automatically configured for you and stored in Azure Key Vault.

Reviewer: Since this section talks about both local and deployed scenarios, the overall App Settings section should be moved on its own. To add additional clarity in the formatting, perhaps bullet the app setting scenarios as:
- Running locally you will need to set some App Settings & User Secrets manually using the [.NET Secret Manager](https://docs.microsoft.com/aspnet/core/security/key-vault-configuration?view=aspnetcore-6.0#secret-storage-in-the-development-environment).
- Deployed to Azure, these secrets are automatically configured for you and stored in the [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/).

Make sure you check out the [readme files](#more-info) in each project's directory for a description of the app settings & secrets you'll need to set in order to run the respective project.

## 2.b. Deploying to Azure - Entire Solution

Deploying to Azure is easy thanks to our pre-configured ARM (Azure Resource Manager) templates.

This button will take you to the Azure portal, passing it the template. You'll be asked a few questions, and then the solution will be up and running in just a few minutes. You will need your B2C configuration values and secrets from step 1.

**Deployment Coming Soon!**
<!-- [![Deploy to Azure](https://www.azuresaas.net/assets/images/deploy-to-azure.svg)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-saas%2Fmain%2Fsrc%2FSaas.Deployment%2FSaas.Deployment.Root%2Fazuredeploy.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-saas%2Fmain%2Fsrc%2FSaas.Deployment%2FSaas.Deployment.Root%2FcreateUiDefinition.json) -->

## 2.c. (Advanced) Deploying to Azure - Single Module

> If you'd like to use just one (or more) module from the project, we've provided bicep templates to do that as well. In each project directory, you'll find a folder named `{ModuleName}.Deployment` that contains all the bicep code you'll need to deploy just that Module. Please be advised that there are certain dependencies that each module requires in order for it to deploy properly. You may find that you need to edit the bicep templates to match your use case. You will find instructions and a list of dependencies for each module within the [module's readme](#more-info).

Reviewer: Since this section indicates that modifying the Bicep templates may be necessary for individual deployment, it it probably a good idea to link to the Bicep documentation here.

## 3. (Optional) Configure Email Provider

> If you choose to deploy the entire solution to Azure, one of the things that gets deployed is the SaaS.Notifications module **need page and link**, an Azure Logic App. This logic app has a simple HTTP trigger that the SaaS.SignupAdministration site is configured to make a POST request to in order to send emails (ie, at the end of the tenant creation process). By default, there is no email provider configured and the emails do not actually get sent. If you'd like to enable email sending, you will need to manually configure your email provider connector of choice inside the Logic App. See the instructions [here](components/saas-notifications.md) to get started.

Reviewer: ```suggestion
The SaaS.Notifications module **need page and link** is an Azure Logic App responsible for generating email notifications. By default, there is no email provider configured. If you'd like to enable email notifications, you will need to manually configure your email provider connector of choice inside the Logic App. See the instructions [here](components/saas-notifications.md) to get started.
```

Recommending we truncate this section and let those be disclosed in components/saas-notifications.md. Since this document is a quick start, details like this can wind up more as noise than signal.

> ## Integrating your application

Reviewer: ```suggestion
## 4. Integrating your application
```

> Now that you've seen how to build and run the code locally, and you have a way to deploy your code to Azure in a repeatable and code-first way, you can integrate your own code into the solution.

Reviewer: ```suggestion
Now that you've seen how to run the code locally as well as deploy your code to Azure (in a repeatable and code-first way), you can integrate your own code into the solution.
```

We've included a basic application within the `Saas.Application.Web` project that demonstrates a SaaS solution called "BadgeMeUp". BadgeMeUp is simply a badge sharing site that *Contoso* (representing your company) can sell to end customers.

> SaaS solutions come in many shapes as sizes. We picked "BadgeMeUp", because it's a fairly simple scenario to understand. [You can read more about this particular SaaS scenario here](../resources/contoso-badgemeup/).

> ### How does this work?

Reviewer: Since this gives further info about Bicep and isn't specific to integration, this section will probably be better suited if moved below 2.b. or 2.c.

Reviewer: Alternatively, this could be changed to a Header2 equal with the sections above and below.
